File: leetcode/java/53.Maximum Subarray.java
public class Solution {
    public int maxSubArray(int[] nums) {
        int currSum = 0;
        int max = Integer.MIN_VALUE;
        for (int i = 0; i < nums.length; i ++) {
            currSum = Math.max(currSum + nums[i], nums[i]);
            max = Math.max(currSum, max);
        }
        return max;
    }
}

File: leetcode/kotlin/53.Maximum Subarray.kt
class Solution {
    fun maxSubArray(nums: IntArray): Int {
        var currSum = 0
        var max: Int = Integer.MIN_VALUE
        for (i in nums.indices) {
            currSum = Math.max(currSum + nums[i], nums[i])
            max = Math.max(currSum, max)
        }
        return max
    }
}